packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -13,81 +13,193 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 /**
  * External dependencies
  */
+import PropTypes from 'prop-types';
 import { useFeature } from 'flagged';
 import styled from 'styled-components';
-import { useCallback, useState } from '@googleforcreators/react';
-import { __ } from '@googleforcreators/i18n';
+import { useCallback, useState, useEffect } from '@googleforcreators/react';
+import { __, sprintf } from '@googleforcreators/i18n';
 import {
-  Button,
-  BUTTON_VARIANTS,
-  BUTTON_TYPES,
-  BUTTON_SIZES,
   Text,
   THEME_CONSTANTS,
+  SearchInput,
+  useLiveRegion,
 } from '@googleforcreators/design-system';
 
 /**
  * Internal dependencies
  */
+import { Section } from '../../common';
+import { useAPI } from '../../../../app';
 import { Row } from '../../../form';
-import useLibrary from '../../useLibrary';
 import { Pane } from '../shared';
+import { useStory } from '../../../../app/story';
+import useLibrary from '../../useLibrary';
+import { getOnPageProducts } from './util';
 import paneId from './paneId';
-import ProductDropdown from './productDropdown';
+import ProductList from './productList';
 
 const HelperText = styled(Text).attrs({
   size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
 })`
   color: ${({ theme }) => theme.colors.fg.secondary};
 `;
+function NoProductsFound({ searchTerm }) {
+  if (searchTerm && searchTerm?.length >= 1) {
+    return <HelperText>{__('No products found.', 'web-stories')}</HelperText>;
+  }
+  return null;
+}
+
+NoProductsFound.propTypes = {
+  searchTerm: PropTypes.string,
+};
 
 function ShoppingPane(props) {
-  const isEnabled = useFeature('shoppingIntegration');
-  const [product, setProduct] = useState(null);
+  const isShoppingIntegrationEnabled = useFeature('shoppingIntegration');
+  const speak = useLiveRegion('assertive');
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isMenuFocused, setIsMenuFocused] = useState(false);
+  const [onPageProducts, setOnPageProducts] = useState([]);
+  const [products, setProducts] = useState([]);
+  const {
+    actions: { getProducts },
+  } = useAPI();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
+
+  useEffect(() => {
+    setOnPageProducts(getOnPageProducts(currentPage));
+  }, [currentPage]);
+
+  const getProductsByQuery = useCallback(
+    async (value = '') => {
+      try {
+        setProducts(await getProducts(value));
+      } catch (err) {
+        setProducts([]);
+      }
+    },
+    [getProducts]
+  );
+
+  const onSearch = useCallback(
```

Comment:
Let's use `useDebouncedCallback` here, with a timeout of 300 should work.

Reply:
What's the correct syntax for this scenario ?

```javascript 
const debouncedSearchChange = useDebouncedCallback(async (value) => {
    await trackEvent('search', {
      search_type: 'dashboard',
      search_term: value,
    });
    setKeyword(value);
  }, TEXT_INPUT_DEBOUNCE);
```

If I use something like this ☝️ I seem to only get back a SyntheticEvent which I can't seem to pull the value from.

Reply:
Don't have an example handy, but there's plenty of precedence in the code base. Prometheus pod might be able to help if needed.